src/shogun/lib/observers/ParameterObserver.cpp

```diff
@@ -72,10 +87,24 @@ bool ParameterObserver::filter(const std::string& param)
 	if (m_observed_parameters.size() == 0)
 		return true;
 
-	for (auto v : m_observed_parameters)
-		if (v == param)
-			return true;
-	return false;
+	auto res = std::find(m_observed_parameters.begin(),
+						 m_observed_parameters.end(),
+						 param);
+
+	return res != m_observed_parameters.end();
+}
+
+bool ParameterObserver::filter(const AnyParameterProperties &property) {
```

Comment:
filter is not a good name, that would imply it returns a (sub)set of parameters, if it returns a bool "has" would be more appropriate or?

Reply:
`has` is already used by SGObject and I would like to avoid confusion. I don't mind the `filter` name since basically it is returning a sub-set of observations (the one we care about).

Reply:
but it returns a bool, or am I wrong?

Reply:
Actually why not `has` again? The class is separate or?

Reply:
yes yes, it returns a boolean in the end. `ParameterObserver` inherits from SGOject and thus it has already an `has` method.

Reply:
I'd suggest `observes` then, as again, `filter` implies that it returns a subset, not a bool

Reply:
`observes` sounds good.